### ISSUE TITLE
Parametric UMAP changing n_components

### ISSUE THREAD

**Issue by jerssonleon** [lmcinnes/umap#586]
Hello,

I'm new with parametric UMAP, also i need to perform dimensionality reduction in a dataset and obtain 8 dimensional feature vector.

I understand that the parameter that i have to chage is n_components 

I already change n_components from 2 to 8

# define the network 
import tensorflow as tf
dims = (28, 28, 1)
n_components = 8
encoder = tf.keras.Sequential([
    tf.keras.layers.InputLayer(input_shape=dims),
    tf.keras.layers.Conv2D(
        filters=32, kernel_size=3, strides=(2, 2), activation="relu", padding="same"
    ),
    tf.keras.layers.Conv2D(
        filters=64, kernel_size=3, strides=(2, 2), activation="relu", padding="same"
    ),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(units=256, activation="relu"),
    tf.keras.layers.Dense(units=256, activation="relu"),
    tf.keras.layers.Dense(units=n_components),
])
encoder.summary()

##############

but an error appears:

/usr/local/lib/python3.6/dist-packages/umap/umap_.py in fit(self, X, y)
   2560             # Might be worth moving this into simplicial_set_embedding or _fit_embed_data
   2561             disconnected_vertices = np.array(self.graph_.sum(axis=1)).flatten() == 0
-> 2562             self.embedding_[disconnected_vertices] = np.full(self.n_components, np.nan)
   2563 
   2564             self.embedding_ = self.embedding_[inverse]

ValueError: shape mismatch: value array of shape (2,) could not be broadcast to indexing result of shape (0,8)

Please help me.

Thank you

**Comment by lmcinnes** [comment]
I think this may actually be a bug introduced by a couple of orthogonal additions that are having a less than ideal interaction.
I'll try to take a look soon.

**Comment by jerssonleon** [comment]
Thank you for your response.

Sincerely,

JL

**Comment by leriomaggio** [comment]
Since I am already working on that part of the code, I would be happy to look into this as well if it's ok for you @lmcinnes 

Right now I assigned the issue to myself. Feel free to change that, in case :)

**Comment by lmcinnes** [comment]
Absolutely; I have been dragged away by other matters at the moment, so if you have time I would greatly appreciate the help.

**Comment by lmcinnes** [comment]
I ended up doing a little digging relating to issue #592 . It is looking less clear to me that this is not something related to ParametricUMAP itself. Perhaps @timsainb could take a look and see if there is anything obvious?

**Comment by timsainb** [comment]
I can take a look. What is the relation to #592 though?

**Comment by timsainb** [comment]
Here is a colab notebook where I copied your code and didn't reproduce the error:
https://colab.research.google.com/drive/1jXi5-cGlIgOkL1sDGndGmYrNVBqIpzsr?usp=sharing

Maybe what happened is that `n_components` was not passed to `ParametricUMAP`? 

We could raise a ValueError in ParametricUMAP when the network architecture has a different latent dimensionality than ParametricUMAP?

**Comment by jerssonleon** [comment]
> Here is a colab notebook where I copied your code and didn't reproduce the error:
> https://colab.research.google.com/drive/1jXi5-cGlIgOkL1sDGndGmYrNVBqIpzsr?usp=sharing
> 
> Maybe what happened is that `n_components` was not passed to `ParametricUMAP`?
> 
> We could raise a ValueError in ParametricUMAP when the network architecture has a different latent dimensionality than ParametricUMAP?

Thank you very much for your colab notebook. I'm going to try with my data.

Sincerely,

JL

**Comment by behroozazarkhalili** [comment]
> > Here is a colab notebook where I copied your code and didn't reproduce the error:
> > https://colab.research.google.com/drive/1jXi5-cGlIgOkL1sDGndGmYrNVBqIpzsr?usp=sharing
> > Maybe what happened is that `n_components` was not passed to `ParametricUMAP`?
> > We could raise a ValueError in ParametricUMAP when the network architecture has a different latent dimensionality than ParametricUMAP?
> 
> Thank you very much for your colab notebook. I'm going to try with my data.
> 
> Sincerely,
> 
> JL

Yeah, it should be better to have raised if n_components has not passed to parametricUMAP fit_transformer method.
